MoneySaver.Api.Services/Implementation/TransactionCategoryService.cs

```diff
@@ -23,10 +23,18 @@ public TransactionCategoryService(IRepository<TransactionCategory> categoryRepos
             this.mapper = mapper;
         }
 
-        public TransactionCategoryModel CreateCategory(TransactionCategoryModel categoryModel)
+        public async Task<TransactionCategoryModel> CreateCategoryAsync(TransactionCategoryModel categoryModel)
         {
-            TransactionCategory transactionCategory = mapper.Map<TransactionCategory>(categoryModel);
-            this.categoryRepository.AddAsync(transactionCategory);
+            try
+            {
+                TransactionCategory transactionCategory = mapper.Map<TransactionCategory>(categoryModel);
+                await this.categoryRepository.AddAsync(transactionCategory);
+            }
+
+            catch (Exception ex)
+            {
+                ;
```

Comment:
You need to return something different ( null for example) if an exception arises.